optuna/storages/_in_memory.py

```diff
@@ -63,7 +65,10 @@ def create_new_study(self, study_name: Optional[str] = None) -> int:
             else:
                 study_uuid = str(uuid.uuid4())
                 study_name = DEFAULT_STUDY_NAME_PREFIX + study_uuid
-            self._studies[study_id] = _StudyInfo(study_name)
+            study = _StudyInfo(study_name)
+            study.directions = list(directions)
+
+            self._studies[study_id] = study
```

Comment:
How about giving `directions` to `_StudyInfo` directly? We need to change the definition of `_StudyInfo`.
```suggestion
            self._studies[study_id] =  _StudyInfo(study_name, list(directions))
```

Reply:
Thanks for the solution!